packages/apollo/src/commands/service/push.ts

```diff
@@ -28,6 +35,13 @@ export default class ServicePush extends ProjectCommand {
       {
         title: "Uploading service to Engine",
         task: async () => {
+          // handle partial schema uploading
+          if (flags.federated) {
+            const info = await (project as GraphQLServiceProject).resolveFederationInfo();
```

Comment:
I think if we leverage the generic types for `this.runTasks` we _might_ be able to get rid of the cast here. See `ClientCheck` or `ServiceCheck` for an example.